tests/components/androidtv/patchers.py

```diff
@@ -0,0 +1,129 @@
+"""Define patches used for androidtv tests."""
+
+from socket import error as socket_error
+from unittest.mock import patch
+
+
+class AdbCommandsFake:
+    """A fake of the `adb.adb_commands.AdbCommands` class."""
+
+    def ConnectDevice(self, *args, **kwargs):  # pylint: disable=invalid-name
```

Comment:
This name is necessary because it needs to match the name in the `adb.adb_commands.AdbCommands` class.  Same goes for the other names below.

Reply:
Another approach would be to spec / autospec a Mock with the correct library class and then set side effects as needed on the mock methods. The benefit with that is that it will raise or fail tests if the library changes and we don't update accordingly.

Reply:
That makes sense, but I don't know how to do that. 

As it is, I structured the `androidtv` package so that all ADB stuff is handled in `adb_manager.py`. In these tests, I'm patching two classes in this module that are imported from two ADB-related packages. If those packages changed, I would need to either update the `androidtv` package or pin its requirements. But for now, `androidtv`'s requirements are actually my forks of those two ADB-related packages with necessary modifications, and unless/until that changes then I can guarantee that these tests won't break.

Reply:
This explains more about autospeccing:
https://docs.python.org/3/library/unittest.mock.html#auto-speccing